core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -422,6 +425,12 @@ private[spark] class MesosClusterScheduler(
       "--driver-cores", desc.cores.toString,
       "--driver-memory", s"${desc.mem}M")
 
+    if (desc.schedulerProperties.get("spark.jars").isDefined) {
+      val jarFiles = desc.schedulerProperties.get("spark.jars").get
+        .split(",").map(uri => uri.substring(uri.lastIndexOf('/') + 1)).mkString(",")
+      options ++= Seq("--jars", jarFiles)
+    }
+
```

Comment:
It's been a while since I wrote any scala, but can't you use `foreach` here instead of the explicit test?
```scala
desc.schedulerProperties.get("spark.jars")
  .foreach(uris => {
    val jarFiles = uris.split(",").map(uri => uri.substring(uri.lastIndexOf('/') + 1)).mkString(",")
    options ++= Seq("--jars", jarFiles)
  })
```